src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py

```diff
@@ -415,7 +416,7 @@ def load_arguments(self, _):
             c.argument('size', help='The VM size to be created. See https://azure.microsoft.com/en-us/pricing/details/virtual-machines/ for size info.')
             c.argument('image', completer=get_urn_aliases_completion_list)
             c.argument('custom_data', help='Custom init script file or text (cloud-init, cloud-config, etc..)', completer=FilesCompleter(), type=file_type)
-            c.argument('secrets', multi_ids_type, help='One or many Key Vault secrets as JSON strings or files via `@{path}` containing `[{ "sourceVault": { "id": "value" }, "vaultCertificates": [{ "certificateUrl": "value", "certificateStore": "cert store name (only on windows)"}] }]`', type=file_type, completer=FilesCompleter())
+            c.argument('secrets', multi_ids_type, options_list=['--secrets', '-s'], help='One or many Key Vault secrets as JSON strings or files via "@{path}" containing [{"sourceVault":{"id":"value"},"vaultCertificates":[{"certificateUrl":"value","certificateStore":"cert store name(only on windows)"}]}]', validator=_validate_secrets)
```

Comment:
_validate_secrets shouldn't be added here but [here (process_vm_create_namespace)](https://github.com/Azure/azure-cli/blob/dev/src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py#L1193), just like [here (process_vmss_create_namespace)](https://github.com/Azure/azure-cli/blob/dev/src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py#L1013).

Reply:
Also why did you remove the file_type?